StraightLinePrograms/src/straightline.jl

```diff
@@ -817,26 +827,39 @@ function compile!(p::SLProgram)
             elseif isuniminus(op)
                 :($rk = -$x)
             else
-                y, idx = cretrieve(j)
-                mininput = max(mininput, idx)
+                y, idy = cretrieve(j)
+                mininput = max(mininput, idy)
                 if isplus(op)
-                    :($rk = $x + $y)
+                    if idx == 0 && isPoly
```

Comment:
I wonder whether `idx == 0` is correct here, as it can be `0` also when the first argument is a constant, which means constants in the slpoly could be overwritten during evaluation. A simple fix might be to change the returned value from `cretrieve` to be `-1` in case it's constant (it won't change the result of the computation of `mininput`).

Reply:
BTW, I had not tested/updated at all this compile functionality during the most recent developments of SLPs, so I'm surprised and glad it still seems to work! (despite the world-age problem caveat)